crates/pgrepr/src/scalar.rs

```diff
@@ -235,17 +296,29 @@ impl Scalar {
                 let nanos = v.timestamp_nanos_opt().unwrap();
                 DfScalar::TimestampNanosecond(Some(nanos), Some(tz.clone()))
             }
+            (Self::Time(v), ArrowType::Time32(TimeUnit::Second)) => {
+                DfScalar::Time32Second(Some(v.num_seconds_from_midnight() as i32))
+            }
+            (Self::Time(v), ArrowType::Time32(TimeUnit::Millisecond)) => {
+                let secs = v.num_seconds_from_midnight() as i32;
+                let sub_millis = (v.nanosecond() / 1_000_000) as i32;
+                let millis = (secs * 1_000) + sub_millis;
+                DfScalar::Time32Millisecond(Some(millis))
+            }
             (Self::Time(v), ArrowType::Time64(TimeUnit::Microsecond)) => {
-                let nanos = get_nanos_from_time(&v);
-                let micros = nanos_to_micros(nanos);
+                let secs = v.num_seconds_from_midnight() as i64;
+                let sub_micros = (v.nanosecond() / 1_000) as i64;
```

Comment:
is this just truncation or does it round?

Reply:
Just truncation. Here, we expect to get data in micros so there shouldn't be any need for rounding. The library doesn't provide a function to get sub micro seconds, so this is the only way...